apis/python/src/tiledbsoma/_multiscale_image.py

```diff
@@ -252,4 +628,82 @@ def reference_level_properties(self) -> "ImageProperties":
         Lifecycle:
             Experimental.
         """
-        raise NotImplementedError()
+        return self._schema.reference_level_properties
+
+
+# TODO: Push down to C++ layer
+@attrs.define
+class MultiscaleImageSchema:
+
+    reference_level_properties: ImageProperties
+    axis_names: Tuple[str, ...]
+    datatype: pa.DataType
+
+    def __attrs_post_init__(self):  # type: ignore[no-untyped-def]
+        ndim = len(self.reference_level_properties.shape)
+        if len(self.axis_names) != ndim:
+            raise ValueError(
+                f"Invalid axis names '{self.axis_names}'. {ndim} axis names must be "
+                f"provided for a multiscale image with image type "
+                f"{self.reference_level_properties.image_type}. "
+            )
+
+    # mypy false positive https://github.com/python/mypy/issues/5313
+    def create_coordinate_space(self) -> CoordinateSpace:
+        return CoordinateSpace(
+            tuple(Axis(name) for name in self.get_coordinate_space_axis_names())  # type: ignore[misc]
+        )
+
+    def get_coordinate_space_axis_names(self) -> Tuple[str, ...]:
+        # TODO: Setting axes and the coordinate space is going to be updated
+        # in a future PR.
+        x_name: Optional[str] = None
+        y_name: Optional[str] = None
+        z_name: Optional[str] = None
+        for axis_name, axis_type in zip(
+            self.axis_names, self.reference_level_properties.image_type
+        ):
+            if axis_type == "X":
+                x_name = axis_name
+            elif axis_type == "Y":
+                y_name = axis_name
+            elif axis_type == "Z":
+                z_name = axis_name
+        assert x_name is not None  # For mypy (already validated)
+        assert y_name is not None  # For mypy (already validated)
+        if z_name is None:
+            return (x_name, y_name)
+        else:
+            return (x_name, y_name, z_name)
+
+    def to_json(self) -> str:
+        type_str = pyarrow_to_carrow_type(self.datatype)
+        return json.dumps(
+            {
+                "name": self.reference_level_properties.name,
+                "image_type": self.reference_level_properties.image_type,
+                "shape": self.reference_level_properties.shape,
+                "axis_names": self.axis_names,
+                "datatype": type_str,
+            }
+        )
+
+    @classmethod
+    def from_json(cls, data: str) -> Self:
+        kwargs = json.loads(data)
+        axis_names = kwargs.pop("axis_names")
+        type_str = kwargs.pop("datatype")
+        _carrow_to_pyarrow = {
+            "c": pa.int8(),
+            "s": pa.int16(),
+            "i": pa.int32(),
+            "l": pa.int64(),
+            "C": pa.uint8(),
+            "S": pa.uint16(),
+            "I": pa.uint32(),
+            "L": pa.uint64(),
+            "f": pa.float32(),
+            "g": pa.float64(),
+        }
+        type = _carrow_to_pyarrow[type_str]
```

Comment:
Should move into `_arrow_types.py` to reside along with `pyarrow_to_carrow_type`